src/Common/Core/Init.php

```diff
@@ -0,0 +1,185 @@
+        if ($lastModifiedTime === false || $this->getContainer()->getParameter('kernel.debug')) {
+            $lastModifiedTime = time();
+        }
+
+        // define as a constant
+        defined('LAST_MODIFIED_TIME') || define('LAST_MODIFIED_TIME', $lastModifiedTime);
+
+        $this->definePaths();
+        $this->defineURLs();
+        $this->setDebugging();
+
+        // require spoon
+        require_once 'spoon/spoon.php';
+    }
+
+    /**
+     * Define paths
+     */
+    protected function definePaths()
+    {
+        // To be implemented in inherited classes
+    }
+
+    /**
+     * Define URLs
+     */
+    protected function defineURLs()
+    {
+        // To be implemented in inherited classes
+    }
+
+    /**
+     * A custom error-handler so we can handle warnings about undefined labels
+     *
+     * @param int    $errorNumber The level of the error raised, as an integer.
+     * @param string $errorString The error message, as a string.
+     * @return null|false
+     */
+    public static function errorHandler($errorNumber, $errorString)
+    {
+        $errorString = (string) $errorString;
+        // is this an undefined index?
+        if (mb_substr_count($errorString, 'Undefined index:') > 0) {
+            // cleanup
+            $index = trim(str_replace('Undefined index:', '', $errorString));
+
+            // get the type
+            $type = mb_substr($index, 0, 3);
+
+            // is the index locale?
+            if (in_array($type, array('act', 'err', 'lbl', 'msg'))) {
+                echo '{$' . $index . '}';
+            } else {
+                // return false, so the standard error handler isn't bypassed
+                return false;
+            }
+        } else {
+            // return false, so the standard error handler isn't bypassed
+            return false;
+        }
+    }
+
+    /**
+     * This method will be called by the Spoon Exception handler and is specific for exceptions thrown in AJAX-actions
+     *
+     * @param object $exception The exception that was thrown.
+     * @param string $output    The output that should be mailed.
+     */
+    public static function exceptionAJAXHandler($exception, $output)
+    {
+        \SpoonHTTP::setHeaders('content-type: application/json');
+        $response = array(
+            'code' => ($exception->getCode() != 0) ? $exception->getCode() : 500,
+            'message' => $exception->getMessage()
+        );
+        echo json_encode($response);
+        exit;
+    }
+
+    /**
+     * This method will be called by the Spoon Exception handler
+     *
+     * @param object $exception The exception that was thrown.
+     * @param string $output    The output that should be mailed.
+     */
+    public static function exceptionHandler($exception, $output)
+    {
+        $output = (string) $output;
+
+        // mail it?
+        if (SPOON_DEBUG_EMAIL != '') {
```

Comment:
Every occurence of ```SPOON_DEBUG_EMAIL``` should be replace by ```$this->getContainer()->getParameter('fork.debug_email')```
See [PR](https://github.com/forkcms/forkcms/pull/1139) for this.